app/components/github_integration/webhooks/utils.py

```diff
@@ -127,6 +140,18 @@ async def send_edit_difference(
     )
 
 
+def _shorten_same_repo_links(
+    origin_repo: RepositoryWebhooks, matchobj: re.Match[str]
+) -> str:
+    if (
+        matchobj.group("owner") == origin_repo.owner.name
+        and matchobj.group("repo") == origin_repo.name
+    ):
```

Comment:
This is still problematic :( GitHub can just omit setting `origin_repo.owner.name`, so I think we should use `.full_name` which is always present:
```suggestion
    owner, _, repo = origin_repo.full_name.partition("/")
    if matchobj["owner"] == owner and matchobj["repo"] == repo:
```